src/common/transformations/include/transformations/rt_info/reduceop_path.hpp

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/node.hpp"
+#include "openvino/core/runtime_attribute.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+
+TRANSFORMATIONS_API void mark_reduceop_path(const std::shared_ptr<Node>& node);
+TRANSFORMATIONS_API bool is_reduceop_path(const std::shared_ptr<const Node>& node);
+
+/**
+ * @ingroup ie_runtime_attr_api
```

Comment:
It is strange that we introduce separate attribute for it. Can they use common marks for precision sensitive nodes.
Or can we extend the current mark?
Also I am not sure that the right place for thic class for here. Let us also move it to mark precision sensitive header?
The current approach is bot good because multiple headers are distributed among different headers and directories. Though these all are about the same transformations

Reply:
Agree, I will collect them in a single file

Reply:
I considered placing this attribute together in a single file with another ones and it looks like this can lead to a ambiguity. 

Yes we use this attribute in marking transformations but it's meaning is different from `disable_fp16_compression` and from `precision_sensitive`. This attribute is needed only to mark nodes are going into `ReduceMean` and `ReduceSum` inputs. Such attribute does not mean that nodes should be\or shouldn't be kept in f32, it's just means that potentially they can be marked to be kept in f32 if there is an `Exp` this goes into `ReduceSum`/`ReduceMean` through `Slice`, `StridedSlice`, `Broadcast`, `Reshape` and so on.

`disable_fp16_compression` definitely and explicitly means this node should not be compressed into f16 and needs to be kept in f32. `precision_sensitive` rt_info attr is placed on inputs and indeed is used to mark shape subgraphs. I think moving `reduceop_path` inside one of them will lead to confusion. 

We already have practice, where similar rt_info attrs are kept separately: e.g.`disable_fp16_compression` and `precision_sensitive` mentioned above. And `reduceop_path` is the only new rt_info attr introduced in this PR. Let's keep it separately where it is now? At least until we merge and get validation reports.

@mvafin how do you think?

Reply:
@pavel-esir, okay I agree to have such attribute but please put this declaration `mark_subgraphs_to_keep_in_mixed_precision.hpp` and into correspondin definition file. This is because this attribute and methods `mark_reduceop_path` are used only by these transformations

Reply:
answered to this question in the discussion above since it's connected with `TRANSFORMATIONS_API `. Let's continue there.